adbms/src/main/scala/de/up/hpi/informationsystems/adbms/record/internal/TypedMapLike.scala

```diff
@@ -0,0 +1,132 @@
+
+  /** Adds a key/value pair to this map, returning a new map.
+    *  @param    kv the key/value pair
+    *  @tparam   T the type of the value in the key/value pair.
+    *  @return   a new `This` with the new binding added to it
+    */
+  def +[T <: V](kv: (K[T], T)): This =
+    ctor(data + kv)
+
+  /** Removes a key from this map, returning a new map.
+    *  @param    k the key to be removed
+    *  @return   a new `This` without a binding for `key`
+    */
+  def -[T <: V](k: K[T]): This =
+    ctor(data - k)
+
+  /** Adds a number of elements provided by a traversable object
+    *  and returns a new collection with the added elements.
+    *
+    *  @param xs      the traversable object consisting of key-value pairs.
+    *  @return        a new `This` with the bindings of this map and those from `xs`.
+    */
+  def ++[T <: V](xs: TypedMapBase[K, T]): This =
+    ctor(data.++(xs.iterator))
+
+  /** Creates a new map from this one by removing all keys of another
+    *  collection.
+    *
+    *  @param xs     the collection containing the removed elements.
+    *  @return a new `This` that contains all key/value bindings of the current map
+    *  except the key/value bindings of `elems`.
+    */
+  def --(xs: TypedMapBase[K, V]): This =
+    ctor(data.--(xs.iterator.map(_._1)))
+
+  /** Retrieves the value which is associated with the given key. This
+    *  method invokes the `default` method of the map if there is no mapping
+    *  from the given key to a value. Unless overridden, the `default` method throws a
+    *  `NoSuchElementException`.
+    *
+    *  @param  k   the key
+    *  @tparam T   the type of the value in the key/value binding
+    *  @return     the value associated with the given key, or the result of the
+    *              map's `default` method, if none exists.
+    */
+  def apply[T <: V : ClassTag](k: K[T]): T =
+    data(k) match {
+      case t: T => t
+      case _ => default(k)
+    }
+
+  /** Optionally returns the value associated with a key.
+    *
+    *  @param  k    the key value
+    *  @tparam T    the type of the value in the key/value binding
+    *  @return an option value containing the value associated with `key` in this map,
+    *          or `None` if none exists.
+    */
+  def get[T <: V : ClassTag](k: K[T]): Option[T] =
+    data.get(k).flatMap{
+      case t: T => Some(t)
+      case _ => None
+    }
+
+  /** A new immutable map containing updating this map with a given key/value mapping.
+    *  @param    key the key
+    *  @param    value the value
+    *  @return   A new map with the new key/value mapping
+    */
+  def updated[T <: V](key: K[T], value: T): This = this + ((key, value))
+
+  /** Defines the default value computation for the map,
+    *  returned when a key is not found
+    *  The method implemented here throws an exception,
+    *  but it might be overridden in subclasses.
+    *
+    *  @param key the given key value for which a binding is missing.
```

Comment:
W.r.t. style: I think keywords should start on column 0, i.e. one space after the `*`. The additional space in the text before this is just because LL108ff are just the continuation of the first line, right?

I would personally probably make a distinction between short description and elaboration anyway. What do you think about this:

```scala
/** Defines the default value or behaviour for missing
  *  keys.
  *
  * The value is returned is returned when a key is not
  * found. The base class implementation here throws
  * a [[NoSuchElementException]] but can be overridden
  * in subclasses.
  *
  * @param  key the given ...
  * @tparam T   the type of the value ...
  * @throws NoSuchElementException if the key was not
  *  found
  */
def default...
```

- The last `@throws` descriptions is not tab-aligned because the exception name is so long.
- A leading extra space always indicates a continuation of the last line.
- The first line of the doc is a short as possible description that is *optionally* followed up with an elaboration that is then, not indented on subsequent lines (because it is a text block).

What do you think? I don't want to lose a lot of time on it but since I think you have put some thoughts into how to write the doc comments I thought I'd chime in to just nail down one style to use from now on.

Reply:
I wouldn't use extra space for continued lines of text. It is manual work not really needed:
- the short description is followed by two newlines
- parameter descriptions etc. are followed by a new annotation which is highlighted by the editor

This is based on the assumption that we don't mix descriptions and annotations. Otherwise I agree :+1: 
```scala
/** Defines the default value or behaviour for missing
  * keys.
  *
  * The value is returned when a key is not
  * found. The base class implementation here throws
  * a [[NoSuchElementException]] but can be overridden
  * in subclasses.
  *
  * @param  key the given ...
  * @tparam T   the type of the value ...
  * @throws NoSuchElementException if the key was not
  * found and some other very long text, bla, bla
  */
def default...
```

Reply:
Can you create a document (in `/doc`) for that if you agree?

Reply:
Sure can. Can we get codacy to check the scaladoc comment layout/style? If there is a preset that kinda matches this setup that would be fastest and most easily enforced.

Reply:
I will check that

Reply:
Cool, thank you. I couldn't find anything just now but I'm not familiar with codacy configuration so it might just be that.

Reply:
I couldn't find anything neither.

Reply:
Ok, thanks for checking though. I opened #104  for this.